src/blockrelay/graphene_set.cpp

```diff
@@ -356,6 +354,134 @@ std::vector<uint64_t> CGrapheneSet::Reconcile(std::set<uint64_t> &receiverSet, c
     return orderedSetItems;
 }
 
+double CGrapheneSet::TruePositiveMargin(uint64_t nSenderFilterPositiveItems,
+    uint64_t nReceiverUniverseItems,
+    double senderBloomFpr,
+    uint64_t nLowerBoundTruePositives)
+{
+    return (nSenderFilterPositiveItems - nLowerBoundTruePositives) /
```

Comment:
You have a few new APIs that / by something derived from input parameters.  I'm just making this comment here, but please search for all cases.

In this case for example if nReceiverUniverseItems == nLowerBoundTruePositives then we will get a DIV by 0 core dump.  Maybe it makes no sense whatsoever for the above to be the case, but we want to armor functions against garbage input.

I'd like to see code that ensures that /0 doesn't happen.  If there's a reasonable value to return in the /0 case, you could do that, or DbgAssert if it could only happen if the caller didn't understand how this function should be called, or raise an exception if it could happen due to legitimate inputs.  If you raise an exception we need to make sure that that exception is caught.  

If the function is called during message processing, you are OK for this because all exceptions are caught at the outer loop and that message is dropped.

Reply:
Good suggestion. I went through the whole file and took care of any cases where there could possibly be division by zero. Fortunately, there are fairly obvious default in those cases, so I didn't need to throw or assert.

Another thing that I did, while I was in there, was to test for [floating point errors](https://en.cppreference.com/w/cpp/numeric/fenv/FE_exceptions) in places where methods like pow, sqrt, and log are used. This should cover cases where I hit overflow or nans.